src/prefect/server/schemas/responses.py

```diff
@@ -303,13 +303,26 @@ class WorkQueueResponse(schemas.core.WorkQueue):
         default=None,
         description="The name of the work pool the work pool resides within.",
     )
+    status: Optional[schemas.statuses.WorkQueueStatus] = Field(
+        default=None, description="The queue status."
+    )
 
     @classmethod
     def from_orm(cls, orm_work_queue):
         response = super().from_orm(orm_work_queue)
         if orm_work_queue.work_pool:
             response.work_pool_name = orm_work_queue.work_pool.name
 
+        if response.is_paused:
+            response.status = schemas.statuses.WorkQueueStatus.PAUSED
+        else:
+            unready_at = datetime.datetime.now(
+                tz=datetime.timezone.utc
+            ) - datetime.timedelta(seconds=WORK_QUEUE_LAST_POLLED_TIMEOUT_SECONDS)
+            if response.last_polled and response.last_polled > unready_at:
+                response.status = schemas.statuses.WorkQueueStatus.READY
+            else:
+                response.status = schemas.statuses.WorkQueueStatus.NOT_READY
```

Comment:
might be nice to have a test case covering the `last_polled <= unready_at` => NOT_READY

Reply:
I concur

Reply:
I've added a test that verifies that conditional branch!